### PR TITLE
Fix typos found by codespell

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
# What has changed ?

Fix most typos found by [codespell](https://github.com/codespell-project/codespell), except in `oxidation` and `tests`: they seem at least partially generated and can be handled in a later merge request if needed.

- [ ] Does this PR affect the User ?

### CODE REVIEW

**Comment by ADcorpo** [comment]
Apart from the small mistake present in our original text (see https://github.com/Scille/parsec-cloud/pull/3695/files#r1039790840 ), looks good to me. No critical functionality is touched and all corrections properly match the original text meaning.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ When proposing a new **feature**, please provide the following:
 2. *why* this important for you
 3. optionally, a proposed solution
 
-> Describing your use case and goals, rather than only / directly jumping into a specific, concrete proposal for a solution is very important. As this allows us to take into account the bigger picture. If you "only" propose concrete solution, and that doesn't fit in, we might need to reverse engineer your actualy use case / goals first to then come up with a solution that fits.
+> Describing your use case and goals, rather than only / directly jumping into a specific, concrete proposal for a solution is very important. As this allows us to take into account the bigger picture. If you "only" propose concrete solution, and that doesn't fit in, we might need to reverse engineer your actually use case / goals first to then come up with a solution that fits.
```
[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply CONTRIBUTING.md]: Indeed, fixed now.

**Review by FirelightFlagboy — APPROVED** [review]
Thank you for your PR and your time :)

File: newsfragments/3700.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix various typos throughout our codebase
```
Comment (FirelightFlagboy) [inline newsfragments/3700.bugfix.rst:1]: It's more a `misc` than a `bugfix`

Comment (ADcorpo) [reply newsfragments/3700.bugfix.rst]: Fixed in 9273ae33c

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**